backend/spec/controllers/spree/admin/orders/customer_details_controller_spec.rb

```diff
@@ -8,7 +8,7 @@
 
     let(:order) { create(:order, number: "R123456789") }
 
-    before { allow(Spree::Order).to receive(:find_by_number!) { order } }
+    before { allow_any_instance_of(Spree::Order::ActiveRecord_Relation).to receive(:find_by!) { order } }
```

Comment:
I had to change the code as shown or the specs would fail. 

The reason behind this is unclear to me, as I didn't expect that changing only the message name (from `:find_by_number!` to `:find_by!`) would make the specs to fail, but that's what happens.

The relevant line for that `before` block is:
```
@order = Spree::Order.includes(:adjustments).find_by_number!(number: params[:order_id])
```

`Spree::Order.includes(:adjustments)` is not an instance of `Spree::Order`, but an instance of `Spree::Order::ActiveRecord_Relation`, so I am puzzled about how it worked before my changes.